pkg/syncer/broker/syncer.go

```diff
@@ -293,6 +293,29 @@ func (s *Syncer) Start(stopCh <-chan struct{}) error {
 		}
 	}
 
+	lister := func(s syncer.Interface) []runtime.Object {
+		list, err := s.ListResources()
+		if err != nil {
+			klog.Errorf("Unable to reconcile - error listing resources: %v", err)
+			return nil
+		}
+
+		return list
+	}
+
+	for i := 0; i < len(s.syncers); i += 2 {
```

Comment:
Ok, I checked that the resource syncers are ready at this point, because you wait for cache sync before exiting the syncer.Start(stopCh) function.